pystiche_papers/ulyanov_et_al_2016/_modules.py

```diff
@@ -188,6 +270,20 @@ def forward(self, *inputs: torch.Tensor) -> torch.Tensor:
 
 
 class BranchBlock(nn.Module):
+    r"""BranchBlock from :cite:`ULVL2016`.
+
+    This block combines the ``deep_branch`` with the ``shallow_branch`` using a
+    :class:`~pystiche_paper.ulyanov_et_al_2016._modules.JoinBlock`.
```

Comment:
Isn't this basically a higher level, i. e. not the first, from the paper? If that is the case, I think its more clear to say `deep_branch` is the previous and `shallow_branch` is the current level.

Reply:
Basically this block combines only these two branches with the help of the `JoinBlock`. Where the branches come from is not known to the block. That's why I'd leave it so general. That the `deep_branch `comes from a previous level is defined in `level`. The naming of the branches comes from the paper.

Reply:
Yeah, but this block has no other purpose. I can't imagine a use-case where it is used otherwise. Furthermore, the names `deep_branch` and `shallow_branch` hint at a very specific purpose. So we should either rename those or make the doc string more specific. I'm voting for the latter. Maybe something like "joins the branch from the previous level with the current"